flink-ml-api/src/main/java/org/apache/flink/ml/api/core/Pipeline.java

```diff
@@ -97,17 +100,17 @@ public PipelineModel fit(Table... inputs) {
     }
 
     @Override
-    public void save(String path) throws IOException {
-        throw new UnsupportedOperationException();
+    public Map<Param<?>, Object> getUserDefinedParamMap() {
+        return paramMap;
     }
 
-    public static Pipeline load(String path) throws IOException {
-        throw new UnsupportedOperationException();
+    @Override
+    public void save(String path) throws IOException {
+        ReadWriteUtils.savePipeline(this, stages, path);
```

Comment:
Another concern is that can we assume that the client can access the stored model data when constructing the JobGraph?  Since you are using `FileReader` in `ReadWriterUtils#loadMetadata()`.

I am not sure but we need to let others know this.